conda_auth/hooks.py

```diff
@@ -1,36 +1,46 @@
 """
 A place to register plugin hooks
 """
-from conda.plugins import CondaAuthHandler, CondaPreCommand, hookimpl
+from conda.plugins import CondaAuthHandler, CondaPreCommand, CondaSubcommand, hookimpl
 
-from . import basic_auth, oauth2
+from .handlers import OAuth2Handler, BasicAuthHandler
+from .handlers.oauth2 import manager as oauth2_manager
+from .handlers.basic_auth import manager as basic_auth_manager
 from .constants import OAUTH2_NAME, HTTP_BASIC_AUTH_NAME
+from .cli import auth_wrapper
+
+
+@hookimpl
+def conda_subcommands():
+    """
+    Registers subcommands
+    """
+    yield CondaSubcommand(
+        name="auth", action=auth_wrapper, summary="Authentication commands for conda"
+    )
 
 
 @hookimpl
 def conda_pre_commands():
+    """
+    Registers pre-command hooks
+    """
     yield CondaPreCommand(
         name=f"{HTTP_BASIC_AUTH_NAME}-collect_credentials",
-        action=basic_auth.collect_credentials,
+        action=basic_auth_manager.get_action_func(),
```

Comment:
As I asked above, can't we use the method directly here?
```suggestion
        action=basic_auth_manager.action_func,
```